src/distribution/cell.py

```diff
@@ -100,6 +100,7 @@ def distribute(self):
         encoded = self.local_generator.encode()
         for neighbour_rank in self.generator_neighbourhood.neighbour_ranks:
             request = self.grid.isend(encoded, neighbour_rank)
+            # TODO change requests list to a dict
```

Comment:
Would be nice to be more explicit about the memory leak here

```suggestion
            # TODO: implement a way to flush this requests to prevent memory leak
```